src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioSymbolNavigationService.cs

```diff
@@ -123,7 +123,13 @@ public bool TryNavigateToSymbol(ISymbol symbol, Project project, OptionSet optio
                 if (allowDecompilation && !project.Solution.Workspace.Options.GetOption(FeatureOnOffOptions.AcceptedDecompilerDisclaimer))
                 {
                     var notificationService = project.Solution.Workspace.Services.GetService<INotificationService>();
-                    allowDecompilation = notificationService.ConfirmMessageBox(ServicesVSResources.Decompiler_Legal_Notice_Message, ServicesVSResources.Decompiler_Legal_Notice_Title, NotificationSeverity.Warning);
+                    var allowDecompilationOrCancel = notificationService.ConfirmMessageBox(ServicesVSResources.Decompiler_Legal_Notice_Message, ServicesVSResources.Decompiler_Legal_Notice_Title, NotificationSeverity.Warning);
+                    if (allowDecompilationOrCancel == null)
+                    {
+                        return false;
```

Comment:
So hitting escape just stops the navigation entirely? Based on the feedback I've seen, did they want the same as no?

Reply:
The feedback was somewhat inconsistent. I opted for 'Cancel' meaning 'Do not proceed with the current navigation operation'.